### DISCUSSION TITLE
How to skip including certain template files?

### DISCUSSION THREAD

**Discussion by archang** [CoBrALab/RABIES#263] [Q&A] [answered]
Does anyone know how to skip using certain template files in the preprocessing package? Specifically, our lab does not have any vascular mask for our rodent atlas. Leaving out the --vascular_mask argument entirely does not work as it will then default to using the vascular mask provided in the docker container.

**Comment by gdevenyi** [comment]
I believe in the current design, vascular masks are required, so you have a few options:

1. Make a vascular mask for your atlas
2. Choose a different atlas (DSURQE for mice, SIGMA for rats, both have vascular masks)
3. Consider where the vascular mask is used, and produce a "fake" one that doesn't affect the final result (this work may be more than 1)

**Reply by archang** [reply]
Thank you Gabriel, we will try out these solutions!

**Comment by Gab-D-G** [comment, marked as answer]
The parameter for each template mask is non-optional (a mask must be provided), but you do not need to provide an 'accurate' mask if you are not using any of those for confound correction (i.e. regression of WM/CSF or vascular signal, or using ICA-AROMA which requires CSF mask). Instead, I recommend providing the brain mask instead as replacement, and the pipeline will run smoothly.

**Reply by archang** [reply]
Thank you for the suggstion to use the brain mask as replacement for the vascular mask, but wouldn't this tell RABIES that we want to regress out the entire brain as vascular signals? I tried to use an empty mask yesterday and got an error along the lines of "The mask is invalid as it is empty: it masks all data" so I am unsure if using a mask of all 0's is the correct approach either.

**Reply by Gab-D-G** [reply]
No regression is applied at the confound correction step unless it is specified to do so. For instance, the signal measured from the vascular mask won't be regressed unless `--conf_list vascular_signal` is specified. So the vascular regression should not be used if you replace it for a whole brain mask, as it would instead regress the whole brain signal, but it should not have any other impact otherwise. Empty masks are not accepted, so it will indeed throw an error, and you should provide a valid mask as replacement.

**Reply by archang** [reply]
This is super helpful - thanks!
